lib/memo_wise.rb

```diff
@@ -3,6 +3,49 @@
 require "memo_wise/version"
 
 module MemoWise
-  class Error < StandardError; end
-  # Your code goes here...
+  def initialize(*values)
+    @_memo = {}
+    super
+  end
+
+  def self.prepended(target)
+    class << target
+      # Implements memoization for the given method name.
+      #
+      # The method must be a *pure function* of the value attributes, and
+      # therefore referentially transparent. Replaces the method with a new
+      # method of the same name, which memoizes the result of calling the
+      # original method. See module comment above for a usage example.
```

Comment:
Perhaps this comment should change (could be in a later PR, just calling it out) as the concept of referential transparency, when I wrote this original comment, was specific to the narrow concern of **adding memoization to a Value object**. 

Given expressed goal for this gem to target memoization in general, not limited to value objects, I think this comment is no longer useful?